tools/envoy_collect/envoy_collect.py

```diff
@@ -0,0 +1,243 @@
+#!/usr/bin/env python
+"""Wrapper for Envoy command-line that collects stats/log/profile.
+
+Example use:
+
+  ./tools/envoy_collect.py --output-path=./envoy.tar -c
+  ./configs/google_com_proxy.json --service-node foo
+  <Ctrl-C>
+  tar -tvf ./envoy.tar
+  -rw------- htuch/eng         0 2017-08-13 21:13 access_0.log
+  -rw------- htuch/eng       876 2017-08-13 21:13 clusters.txt
+  -rw------- htuch/eng        19 2017-08-13 21:13 listeners.txt
+  -rw------- htuch/eng        70 2017-08-13 21:13 server_info.txt
+  -rw------- htuch/eng      8443 2017-08-13 21:13 stats.txt
+  -rw------- htuch/eng      1551 2017-08-13 21:13 config.json
+  -rw------- htuch/eng     32681 2017-08-13 21:13 envoy.log
+
+The Envoy process will execute as normal and will terminate when interrupted
+with SIGINT (ctrl-c on stdin), collecting the various stats/log/profile in the
+--output-path tarball.
+
+TODO(htuch):
+  - Generate the full perf trace as well, since we may have a different version
+    of perf local vs. remote.
+  - Writeup some MD docs for GitHub.
+  - Add a Bazel run wrapper.
+  - Flamegraph generation in post-processing.
+  - Support other modes of data collection (e.g. snapshotting on SIGUSR,
+    periodic).
+  - Validate in performance mode that we're using an opt binary.
+  - Consider handling other signals.
+  - bz2 compress tarball.
+  - Use freeze or something similar to build a static binary with embedded
+    Python, ending need to have Python on remote host (and care about version).
+"""
+from __future__ import print_function
+
+import argparse
+import ctypes
+import ctypes.util
+import datetime
+import json
+import os
+import pipes
+import shutil
+import signal
+import subprocess as sp
+import sys
+import tarfile
+import tempfile
+from six.moves import urllib
+
+DEFAULT_ENVOY_PATH = os.getenv('ENVOY_PATH',
+                               'bazel-bin/source/exe/envoy-static')
+PERF_PATH = os.getenv('PERF_PATH', 'perf')
+
+PR_SET_PDEATHSIG = 1  # See prtcl(2).
+
+DUMP_HANDLERS = ['clusters', 'listeners', 'server_info', 'stats']
+
+
+def FetchUrl(url):
```

Comment:
can we stick to pep8 and google style on method naming? `method_name`

https://google.github.io/styleguide/pyguide.html?showone=Naming#Naming